### PR TITLE
Calibration creation date

### LINKED ISSUE #145: Add save/load mechanism for pose adjustment in the momentum corrector workflow
Add the possibility to apply the pose adjustment even without pre-binning data from config values, and a corresponding saving mechanism.

### LINKED ISSUE #149: Creation date storing for claibration data
Add a creation_date entry to the calibration/correction entries in the config, and also add this to the meta data upon application of these corrections/calibrations, to track which calibration/correction has been applied.

### PR DESCRIPTION
**Pull request by rettigl**
This PR:
- unifies the interface for calibrations, corrections and offsets, and their saving and loading.
- adds time-stamps for the creation of the respective correction, and displays them when loading (optional).
- adds Saving/loading procedures for momentum transformations/pose adjustments 

closes #145 
closes #149

### CODE REVIEW

**Comment by rettigl** [comment]
This has grown relatively large again, but should be well covered by tests. @steinnymir please inspect the changed structure of the offset functions in energy and delay correctors (functionality should be the same and/or fixed according to docstring)

File: sed/core/processor.py [hunk 1]
```diff
@@ -1545,9 +1765,23 @@ def save_delay_calibration(
...
+        calibration = {}
+        for key, value in self.dc.calibration.items():
+            if key == "datafile":
+                calibration[key] = value
+            elif key in ["adc_range", "delay_range", "delay_range_mm"]:
+                calibration[key] = [float(i) for i in value]
+            else:
+                calibration[key] = float(value)
+
+        if "creation_date" not in calibration:
```
Comment (zain-sohail) [inline sed/core/processor.py:1779]: I just looked at this one example of save_delay_calibration and can't seem to get the intended results. 

Firstly, the self.dc.calibration shows empty after running calibrate_delay_axis (using tutorial notebook 2) but no ValueError is raised.
And hence also the creation_date is not being stored in the config file. Actually tried for others and seems to be the same problem. Is it something I am doing wrong?

Comment (rettigl) [reply sed/core/processor.py:1779]: What exactly did you do/check? The calibration is stored as processor.dc.calibration.
See lines 119 ff. and 193 in delay.py
They are only written to the config if you call the respective save_... function in the processor

Comment (rettigl) [reply sed/core/processor.py:1779]: This is what I get:
image

Comment (zain-sohail) [reply sed/core/processor.py:1779]: image

I am getting this

Comment (rettigl) [reply sed/core/processor.py:1779]: I can't reproduce this. Are you sure you are on the right branch? Maybe try to debug and see if you come by the lines of code I indicated.

Comment (rettigl) [reply sed/core/processor.py:1779]: @zain-sohail Do you need further input to test/debug? I tested once more and still cannot reproduce what you report.

Comment (zain-sohail) [reply sed/core/processor.py:1779]: Sorry for late reply. There was some problem with the install and I managed to check it and everything looks good.

**Review by zain-sohail — APPROVED** [review]

**Pull request merged**